website/docs/how-to/how-to-troubleshoot-flag-exposure.md

```diff
@@ -0,0 +1,9 @@
+---
+title: My feature flag is enabled but all/some of our users are not exposed to it
+---
+
+To confirm how users will have flags resolved, follow these steps:
+1. The Unleash Playground was developed with this particular use case in mind. An access token can be used along with context values (passed in via the UI) to see how a flag will be resolved. Check our [Unleash Playground documentation](../reference/playground.mdx) for more info. 
+2. When using a **gradual rollout** strategy, be mindful of the **"stickiness"** value. The stickiness value is a context value used when determining where a context will fall in a gradual rollout. It is a **required field** for the strategy; satisfying the constraints and strategy requirements alone (without the stickiness value) will result in the strategy not being satisfied causing the Unleash platform to defer to other strategies (if possible) or as a disabled state if no other strategies are satisfied.
+    - Sometimes, different implementations of a feature (for example, front-end and back-end) may be unable to provide the same information or define the information differently leading to different outcomes from the platform.
```

Comment:
This is a little unclear to me. Are we saying it's mandatory when defining the strategy? As far as I remember, you can't set the stickiness value to **no** context field. And if it's 100%, shouldn't it still evaluate to true regardless of the context? I think some SDKs might treat this differently, but I think that's the goal.

Reply:
Yeah, a bit unclear to me as well, let me clarify some stuff and post a suggestion.

> you can't set the stickiness value to **no** context field

You can use the `random` strategy for no context field at all, but because the behavior is random, our `default` strategy will try to use some context fields, if available:
![image](https://github.com/Unleash/unleash/assets/455064/ba9ab8ed-6917-45d2-a85f-242bc1a63501)

But you can still pick random. All other strategies have to define a context field, but random is a predefined/special one that comes with Unleash

I think it's better to keep this sentence short and link to the reference documentation to avoid these two pieces to drift apart from each other
```suggestion
2. When using a **gradual rollout** strategy, be mindful of the **[stickiness](../reference/stickiness)** value. When evaluating a flag, if the provided context does not include the field used in the stickiness configuration, the gradual rollout strategy will be evaluated to false.
```

Reply:
Wait, so does `random` use any stickiness fields?

I like your suggestion, maybe something like this (requires a little more work):
```suggestion
. When using a **gradual rollout** strategy, be mindful of the **[stickiness](../reference/stickiness)** value. When evaluating a flag, if the provided context does not include the field used in the stickiness configuration, the gradual rollout strategy will be evaluated to false. Note that the `random` stickiness is a special case that [... what?]. This also means that if you have defined a context field called `random`, you can't use that for stickiness [is that true?].
```

Reply:
> Wait, so does random use any stickiness fields?

No, random is the only one that does not rely on the context fields at all. So if you're using random as stickiness (or default that ultimately evaluates as random), you will not be facing this issue. Therefore, I think not mentioning these strategies is better to avoid confusion 

> This also means that if you have defined a context field called `random`, you can't use that for stickiness [is that true?]

It shouldn't be true, and if it is, then it's probably a bug. The strategy name, should be used to first try to load a strategy implementation by name, and if no strategy is found, then we use the name as the field we want to be sticky on. But this means that you can't create a context field `random` and be sticky on it (you can but it won't work because random strategy would take precedence)... In any case, I don't think this should be part of this issue as it sounds very edgy (eventually we'll receive a bug report for this :shrug:)

I'm sticking with my suggestion if you're ok:
```suggestion
2. When using a **gradual rollout** strategy, be mindful of the **[stickiness](../reference/stickiness)** value. When evaluating a flag, if the provided context does not include the field used in the stickiness configuration, the gradual rollout strategy will be evaluated to false.```